aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,443 @@ uid: fundamentals/servers/kestrel
+A reverse proxy scenario exists when there are multiple apps that share the same IP and port running on a single server. Kestrel doesn't support this scenario because Kestrel doesn't support sharing the same IP and port among multiple processes. When Kestrel is configured to listen on a port, Kestrel handles all of the traffic for that port regardless of requests' host header. A reverse proxy that can share ports has the ability to forward requests to Kestrel on a unique IP and port.
 
-Even if a reverse proxy server isn't required, using one might be a good choice for other reasons:
+Even if a reverse proxy server isn't required, using a reverse proxy server might be a good choice:
 
-* It can limit your exposed surface area.
-* It provides an optional additional layer of configuration and defense.
+* It can limit the exposed public surface area of the apps that it hosts.
+* It provides an additional layer of configuration and defense.
 * It might integrate better with existing infrastructure.
-* It simplifies load balancing and SSL set-up. Only your reverse proxy server requires an SSL certificate, and that server can communicate with your application servers on the internal network using plain HTTP.
+* It simplifies load balancing and SSL configuration. Only the reverse proxy server requires an SSL certificate, and that server can communicate with your app servers on the internal network using plain HTTP.
 
 > [!WARNING]
-> If you're not using a reverse proxy with host filtering enabled, you must enable [host filtering](#host-filtering).
+> If not using a reverse proxy with host filtering enabled, [host filtering](#host-filtering) must be enabled.
 
 ## How to use Kestrel in ASP.NET Core apps
 
 #### [ASP.NET Core 2.x](#tab/aspnetcore2x/)
+
 The [Microsoft.AspNetCore.Server.Kestrel](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.Kestrel/) package is included in the [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage).
 
-ASP.NET Core project templates use Kestrel by default. In *Program.cs*, the template code calls `CreateDefaultBuilder`, which calls [UseKestrel](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.hosting.webhostbuilderkestrelextensions#Microsoft_AspNetCore_Hosting_WebHostBuilderKestrelExtensions_UseKestrel_Microsoft_AspNetCore_Hosting_IWebHostBuilder_) behind the scenes.
+ASP.NET Core project templates use Kestrel by default. In *Program.cs*, the template code calls [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder), which calls [UseKestrel](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderkestrelextensions.usekestrel) behind the scenes.
```

Comment:
Why even show an example that calls UseKestrel in this scenario?

Reply:
At this point, it doesn't look like (Tom?) was trying to show that ... I think he's going for an example of the ...

> ASP.NET Core project templates use Kestrel by default ... `CreateDefaultBuilder` ... calls `UseKestrel`

... bit. It looks like he's just making reference to the `CreateDefaultBuilder` line at this point.

Then, it looks like he wanted to show the options config, either to show the `ListenOptions` and/or a cert being used ... or perhaps just for some context early in the topic.

I'm just guessing, of course.

Would you like me to strike the `UseKestrel`?

I should mention that when 2.1 ships that this topic is going to get a *green dino samples treatment*: https://github.com/aspnet/Docs/issues/5495. For now, I'm just trying to work with what we have as best I can. We aren't going to have 2.0 and 2.1 samples ... only 2.x ... and 2.x = 2.0 now and 2.x = 2.1 at 2.1 RTM.

Reply:
Removing UseKestrel would be more clear here, showing that the app can run without it.